### PR TITLE
Implement texture mipmapping in `ModelExperimental`

### PR DESCRIPTION
**Pull request by jjhembd**
The TextureManager in ModelExperimental has been updated to generate texture mipmaps (when requested). This resolves https://github.com/CesiumGS/cesium/issues/10231.

WebGL1 can only generate mipmaps for textures with power-of-two dimensions. To handle non-power-of-two textures, some helper functions were added to `Source/Core`:
- `resizeImageToNextPowerOfTwo` for textures supplied as an HTML Image or Canvas
- `getImageFromTypedArray` to convert raw data textures to images for resizing

Note that some textures supplied as `TypedArray`s cannot be resized. In this case `createTexture` simply prints a warning and returns a texture without mipmaps.

The above flow is also used for texture samplers with repeating boundary conditions.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @jjhembd!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/texture-mipmaps/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

**Review by j9liu — COMMENTED** [review]
@jjhembd this is very close! Just one suggestion before this gets merged.

For the record, I put together a [localhost Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=7Vhtc9u4Ef4rOHdujm5oiJLll9BOWscvaTrRnSd2mplGGR9EQhISEOABoG1dx/+9C4AvoKRcmw831w/1WC8EFruLZx8sdnVONasKfPlYUsUKKgzhV5SYSlGNqSAzTicypzycRy+QURU9mYpMCm3QPaMPVMGooA/o3Ov7hxuLpjuZez6XsJAJqqY7MfrXVCAkVU7VG5HTksKbMLeKCM2rjIpslaI54ZrGU/G0C1amwlvAGZfZF5xVStkF4A3YrO39veKMiAtiKJ4rWbzR8vgwGUbW0nRnlIyGe8PhXvL8dpSko6N0/wgnw8PkcDw+OtgfHx8ej5/vH/1zujMV3t5ggNyuUSk1M0wKJhboxbf9WT0en0ZJ5+05UQa+EbHvvL2gC0Wpdt7uDUf7ODkajw+Hz2M7MB7j5CDZP0oO3WPiffSal+Ua7H+jJAdfr5nJlu8k51ESI/vfLZmzR5pfKVJQh/hcqqLzqx3SGKAGJjg5edWueU0hhsRIVSMrpDJLiKl/eqDa9EA8r7SRBbpZEgg2uqBzJhwSGn0LiqCoIF8oIihb0uwLVTNJVI4MfbQ0RQ/MLBERiPBySZBZEoOYyBQlmmo/aZbUackZWUgBBBZVMQMuinklMhcZq/88UH7rdUe1jRv2K41RwcqClCXgu+s57BHtOeWj8Z4Jc3ymFFlFY/RnFGjpP3mgGkUFebxoPHyBRiAa2kd7aGjlEYIAoYhT2CaIJSfwcRoqhYFnz2oPA+HPXvjzuvDnQLjxhMGRfAR5tub8M/T5pJG04bU7t0cDAEYlUcyskJy7pzWo0YLdQzT6EQw0lcQYqgTuu5F3aLCebTZHUTv5PUAFTAGSd9twSjXs2vqif6kI8ETRHKMPFEnBVxAlCgySrQxMgm9ECMq/65SE3n60kXTAfLLROThovXkKNiJBmXpgGmhJf8jXXQCDM04gh6FXlUHEEgRAgQnCga/5KlDkTgrhgLrzM9FISyRAvcXbeg/jYIpiz6B6lY22tefPQr0hYC4pt0YFFH+E9ACb+dQiv3XLgP6+3XYEojDW6NlFg5C1NSIOD/cWcBv2M2cZsaftinHj7ovuQKG/dKA3ucjzbrKxEP94efbu8ub2bvLmenJ2fVc/ovTbVTiHFQUhEWbRet17WAa5MKpZZVYlzd2hTnsoxX76geVmmYbHpZ5YUrZYmm0zELALKX6w5w2cKiWH6ysG4qAHIiCvKKbLfnah+YJqv3YT0HTLWG2oIItN6TWQNkUalJwSdxM/WbI1N0m2mS9tTL+SSY9jf6l3N9GW9R8gW09YqX9bjy0+fOZ0Wf09HDZL7h5SwXlGxCaegugv2yxPYLy+n3o3qb+6/IxnQOXpoNMmzVR3oaK0yz6WKS2+NYtuYQzfnE2u316+uxtdxI3sPeEVTbeBUYs8efjd+1yRha3BvFtWLkU/+2MGmiTLW4kJVFvRVf3wRpSQbubafcbANgmP2a/FXWGLnAmwTkGqAYj8l12vL0im9zQbWf6eS+nuuFoVhqyt2KyCWgY3s3fJSW/deP1+rI/B6CLq4xe3BnYDDY1POGfzeaUdw9hjtD4cW1P7UYKT3bhnEKvdMCsh9HNXVf4WGQIm/k+RovHr/+T4nclhczMtpP7Wsv+rXYCD0/LJQTcjnEspUqiaBzclzfQAehcycF2HHrzmZv5W2jAOXsnHOvT5YMFvr8IBvAA5V30/OZ9tmWdLA30BsbWRsAm3zbe53w76aO03d5pjyXTnWsmM5pWCQNfK66IeGjWhKacZiLXVchRUWn7S+R35LeJ6b12ktvnUZ+8n539rIFRaKV6bq7tADT0ixSV0o1AhQVWJFWzsnp5Bt+NNBniDwa+tInkeeTfqM7nR57rezIYiancLzqT2rT2pWZAR0v+QS9pFzclS7DHd0not15q4sAGLOvY3bWXcDUE/GG9UQZecg3E4//jD65vjcSCwpRdsJndbMQ19IOSiYdKkHJcBPNL2BQflLM+hAXOlb01siKEtRTNZ2LZAS1ejWno2To/+q2a4a4dr42vNsGuHQ2f++GD/TkEe/WFRtp9uL9i1KJApCmhvMARbRF1GcBJtWmh+rwGbiuA5X93KV7ISdsM3pe1avDye9QbjLqvYVGQVpyjBB61LJ2EhekNEnhFtOLXRvZWSz4iaUFFFLtFZqZ1451SbFacvG71/ZUUplbFRjTAeGFqUtujWg1kFsTQ407pLW38yXml42c2ge1so63aK1GJGovEoRs0rwcfh9VSCX7C5FI3Lx2B45n762lMQ/0r3J582TDN3O/duW2WgSud70BsuhK3585zTTat7RpYpGvUsN1MzaYDH/dna9OkgBOw0Z/eI5S+2/IyHMk60hpl5xbntaaY7L08HIL+xlEvH9J/Ac05WVmw5fPnWD2KMTwfwuH1lDcKa5n8D) (adapted from the Custom Shaders Model Sandcastle) to test the mipmapping behavior, and it works! The left cube is with mipmapping; the right is without. The mipmapping effect can be seen on top of the cube.

mipmap

File: Source/Renderer/Texture.js [hunk 1]
```diff
@@ -995,15 +995,17 @@ Texture.prototype.generateMipmap = function (hint) {
...
-  if (this._width > 1 && !CesiumMath.isPowerOfTwo(this._width)) {
-    throw new DeveloperError(
-      "width must be a power of two to call generateMipmap()."
-    );
-  }
-  if (this._height > 1 && !CesiumMath.isPowerOfTwo(this._height)) {
-    throw new DeveloperError(
-      "height must be a power of two to call generateMipmap()."
-    );
+  if (!this._context.webgl2) {
```
[the remark this answers is not in the document]
Comment (jjhembd) [reply Source/Renderer/Texture.js:998]: I updated this documentation to clarify that the power-of-two limitation only applies to WebGL1. Thanks!

**Comment by j9liu** [comment]
Thanks @jjhembd ! Merging now.

**Pull request merged**